Attach verbose logging level to request context

andrewemeryanz opened the pull request:
Internally, sysl-go uses a value attached to the context to
track whether or not additional debug/verbose-level information is
attached to its logs. In order for this value to persist within a
request, it must also be attached to the request context.

Closes #150

tiagralee commented on logconfig/logconfig.go:
```diff
@@ -0,0 +1,30 @@
+
+// Set against the context whether or not sysl-go internals log additional debug/verbose level information.
+func SetVerboseLogging(ctx context.Context, verbose bool) context.Context {
+	ctx = log.WithConfigs(log.SetVerboseMode(verbose)).Onto(ctx)
```
set the verbose mode to context as well, otherwise, the `field.Debug` cannot be logged.

andrewemeryanz replied:
I don't know if this is required. The `pkg` logger stored against the old context should already be `verbose`. If the state isn't being transferred correctly then you might want to dig into the interaction between `log.From` and `log.WithLogger`.

Commit:
Attach verbose logging level to request context

Internally, sysl-go uses a value attached to the context to
track whether or not additional debug/verbose-level information is
attached to its logs. In order for this value to persist within a
request, it must also be attached to the request context.

Closes #150

andrewemeryanz reviewed (commented):
Looks good.

tiagralee commented:
LGTM

tiagralee reviewed (approved):
LGTM

uynap commented:
Thanks Andrew & Alex. I will test it.
